src/utilities/hypre_printf.c

```diff
@@ -116,6 +116,20 @@ free_format( char *newformat )
    return 0;
 }
 
+HYPRE_Int
+hypre_ndigits( HYPRE_BigInt number )
```

Comment:
I'm not sure about the location of this function.  Everything in this file is related to standard C printf facilities.

Reply:
Thanks for the comment @rfalgout, I was not sure either about the location and ended up putting it there since its usage is meant to help the printf functions.